Gems/Terrain/Code/Source/Components/TerrainWorldDebuggerComponent.h

```diff
@@ -112,10 +121,15 @@ namespace Terrain
         static constexpr int32_t MaxVerticesToDraw = 500000;
         static constexpr int32_t MaxSectorsToDraw = MaxVerticesToDraw / VerticesPerSector;
 
-        // Structure to keep track of all our current wireframe sectors, so that we don't have to recalculate them every frame.
-        AZStd::vector<WireframeSector> m_wireframeSectors;
-
         // The size in sectors of our wireframe grid in each direction (i.e. a 5 x 5 sector grid has a sectorGridSize of 5)
-        int32_t m_sectorGridSize{ 0 };
+        // Given the AuxGeom vertex limits, MaxSectorsToDraw is the max number of wireframe sectors we can draw without exceeding the
+        // limits.  Since we want an N x N sector grid, take the square root to get the number of sectors in each direction.
+        // We're always going to keep the camera in the center square, so "round" downwards to an odd number of sectors if we currently
+        // have an even number.  (If we added a sector, we'll go above the max sectors that we can draw with our vertex limits)
+        static constexpr int32_t SqrtMaxSectorsToDraw = 35; // = aznumeric_cast<int32_t>(sqrtf(MaxSectorsToDraw)); (sqrtf is not constexpr)
+        static constexpr int32_t SectorGridSize = (SqrtMaxSectorsToDraw & 0x01) ? SqrtMaxSectorsToDraw : SqrtMaxSectorsToDraw - 1;
+
+        // Structure to keep track of all our current wireframe sectors, so that we don't have to recalculate them every frame.
+        AZStd::array<WireframeSector, SectorGridSize * SectorGridSize> m_wireframeSectors;
```

Comment:
Can we leave this as a vector instead of an array? I'd eventually like to make the size a setting on the component.

Reply:
We cannot unfortunately, because WireframeSector now contains a mutex member variable which is non-copyable.

Reply:
Hrm. What about making the mutex member a unique_ptr<mutex>? That should be copyable.
One of my reasons for wanting to make this a dynamic setting is because the actual distance you can draw depends on the number of other things you put in your world. So there are reasons to go both higher and lower than what it's currently set to. It just hasn't been exposed outwards yet.

Reply:
I tried that originally, but it didn't work (can't remember why). Are you fine with leaving it as an array for now so I can get all this stuff in, then someone can figure out how to change it back to a dynamic container when (if) the work you're describing happens?

Reply:
Yeah, I guess so. It's just a bit frustrating because that someone will be me, and it will probably be in the next month or two. :(

Reply:
We spoke about this in another channel, and resolved the original problem by adding custom copy/assignment operators to the WireframeSector struct, so were able to change this back to being a vector.